lib/glimmer/tk/widget_proxy.rb

```diff
@@ -551,21 +562,91 @@ def handle_listener(listener_name, &listener)
       def on(listener_name, &listener)
         handle_listener(listener_name, &listener)
       end
-      
+
+      def raise_event(event_name, data = nil)
+        data = YAML.dump(data) if data && !data.is_a?(String)
+        tk.event_generate("<#{event_name}>", data: data)
+      end
+
       def unbind_all
         @listeners&.keys&.each do |key|
           if key.to_s.downcase.include?('command')
             @tk.send(key, '')
           else
-            @tk.bind(key, '')
+            @tk.bind_remove(key)
           end
         end
+        @listeners = nil
       end
-      
+
+      def clear
+        unbind_all
+        children.each(&:destroy)
+        @children = []
+      end
+
       def content(&block)
         Glimmer::DSL::Engine.add_content(self, Glimmer::DSL::Tk::WidgetExpression.new, keyword, *args, &block)
       end
 
+      def window?
+        false
+      end
+
+      def closest_window
+        widget = self
+        widget = widget.parent_proxy until widget.window?
+        widget
+      end
+
+      def close_window
+        closest_window.destroy
+      end
```

Comment:
This behavior is not the responsibility of a widget, yet its parent window. It is OK if we write code like `widget.toplevel_parent_proxy.destroy` to get this work done. 

I would rather not introduce this method as it places the responsibility in the wrong place from an object-oriented design point of view, and it adds extra unnecessary code for not much benefit.

Please leave out.